src/sentry/api/endpoints/group_events.py

```diff
@@ -36,47 +36,71 @@ def get(self, request, group):
         :auth: required
         """
 
-        events = Event.objects.filter(
-            group_id=group.id,
-        )
-
-        query = request.GET.get('query')
-
-        if query:
+        def respond(queryset):
```

Comment:
This entire method seems super clunky now, but I couldn't quickly think of a better way to structure it. Suggestions welcome